checkstyle-tester/diff.groovy

```diff
@@ -0,0 +1,243 @@
+import java.nio.file.Paths
+
+import static java.lang.System.err
+
+static void main(String[] args) {
+    def cli = new CliBuilder(usage:'groovy diff.groovy [options]')
+    cli.with {
+        r(longOpt: 'localGitRepo', args: 1, argName: 'localGitRepo', 'Path to local git repository')
+        b(longOpt: 'baseBranch', args: 1, argName: 'baseBranch', 'Base branch name. Default is master')
+        p(longOpt: 'patchBranch', args: 1, argName: 'patchBranch', 'Name of the patch branch in local git repository')
+        c(longOpt: 'checkstyleCfg', args: 1, argName: 'checkstyleCfg', 'Path to checkstyle config file')
+        l(longOpt: 'projectsToTestOn', args: 1, argName: 'projectsToTestOn', 'Path to file which contains projects to test on')
+    }
+    def options = cli.parse(args)
+
+    if (areValidCliArgs(options)) {
+        def localGitRepo = new File(options.localGitRepo)
+        if (hasUnstagedChanges(localGitRepo)) {
+            throw new IllegalStateException("Error: git repository ${localGitRepo.getPath()} has unstaged changes!")
+        }
+
+        def baseBranch = options.baseBranch
+        if (!baseBranch) {
+            baseBranch = 'master'
```

Comment:
@romani Users can modify their local branches and put in any commits they want and we won't be able to tell the difference. Reports don't mention commit hashes they are working with. They are also not required to keep it up to date. With remote master (`origin/master`) and fetching, we know we are always dealing with the latest changes and what everyone else is using.